### PR TITLE
Fixed generating Dart structs with freezed + methods

### PR DESCRIPTION
**Pull request by anlumo**
#969

## Checklist

- [X] An issue to be fixed by this PR is listed above.
- [X] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [X] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [X] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job! Feel free to ping me when ready for review

**Comment by anlumo** [comment]
Yeah, it's just the tests that need fixing. I hope that I can get to it soon.

**Comment by fzyzcjy** [comment]
Take your time!

**Comment by anlumo** [comment]
Should be done now! I don't know what the error in the CI is there, doesn't seem to be related to my change (missing readme?).

**Comment by fzyzcjy** [comment]
I have submitted a fix just now (http -> https), no worries about that failure.

File: frb_example/pure_dart/rust/src/api.rs [hunk 1]
```diff
@@ -1320,3 +1320,15 @@ pub fn handle_type_alias_model(input: Id) -> TestModel {
...
     }
 }
+
+#[frb(dart_metadata = ("freezed"))]
+#[derive(Clone)]
+pub struct FreezedPlusFunction {
+    pub text: String,
+}
+
+impl FreezedPlusFunction {
```
Comment (fzyzcjy) [inline frb_example/pure_dart/rust/src/api.rs:1330]:
```suggestion
impl FreezedPlusMethod {
```

File: frb_example/pure_dart/rust/src/api.rs [hunk 2]
```diff
@@ -1320,3 +1320,15 @@ pub fn handle_type_alias_model(input: Id) -> TestModel {
...
+    pub fn as_string(&self) -> String {
```
Comment (fzyzcjy) [inline frb_example/pure_dart/rust/src/api.rs:1331]: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

```suggestion
    pub fn as_str(&self) -> &str {
```

or `pub fn inner(&self) -> String` etc

**Review by fzyzcjy — COMMENTED** [review]
Ready to merge and good job :) Just naming uber-nits

**Comment by anlumo** [comment]
Why wasn't this merged yet? Any issues?

**Comment by fzyzcjy** [comment]
Oops I did not get any notifications! (github just keep giving notifications about each commit and thus the really useful comments are submerged) Feel free to ping me if I do not reply.

So, we may need to ensure CI passes before we can merge it in this PR

**Comment by anlumo** [comment]
Ignore the closing, misclicked on the GitHub UI.

**Comment by anlumo** [comment]
Small update: I got reassigned to a different project for the time being, so I can't continue my work for this.

**Comment by fzyzcjy** [comment]
Well done!

**Comment by fzyzcjy** [comment]
@all-contributors please add @anlumo for code

**Pull request merged**